apis/kueue/v1alpha1/workload_webhook.go

```diff
@@ -58,15 +59,29 @@ var _ webhook.Validator = &Workload{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Workload) ValidateCreate() error {
-	return nil
+	return validateWorkload(r).ToAggregate()
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Workload) ValidateUpdate(old runtime.Object) error {
-	return nil
+	return validateWorkload(r).ToAggregate()
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
 func (r *Workload) ValidateDelete() error {
 	return nil
 }
+
+func validateWorkload(obj *Workload) field.ErrorList {
```

Comment:
We should also validate the `PriorityClassName`, it should be a DNS subdomain, although we ensured this in JobController, but we will have multi implements in the future, it can prevent the problem as early as possible.

Reply:
Should we also make sure the queue name shouldn't be empty as I asked before. I think so if we don't have default ones. @ahg-g

Reply:
> Should we also make sure the queue name shouldn't be empty as I asked before. I think so if we don't have default ones.

How about if the user create the default queue after the workload created ?

Reply:
I think we should not fail the Workload creation if the queue name is missing. This is because a user could create a Job without a queue annotation and decide to add it later. We need the Workload object to post an Event that indicates why the workload is not schedulable.